### ISSUE TITLE
Generated application does not correctly access config

### ISSUE THREAD

**Issue by dmlloyd** [kiegroup/kogito-runtimes#256]
In the (I assume) generated `org.kie.kogito.app.Application` class, there is a configuration property defined like this:

```java
    @org.eclipse.microprofile.config.inject.ConfigProperty(name = "kogito.service.url", defaultValue = "")
    java.lang.String kogitoService;
```

However, per the SmallRye Config implementation and ongoing spec discussions, the behavioral expectation of this construct (populating the field with an empty string if the value is not found) is incorrect.  The field should be declared as an `Optional<String>`, and the `defaultValue` is redundant and can be left off.

This blocks quarkusio/quarkus#6119.

**Comment by mswiderski** [comment]
@dmlloyd will work on it tomorrow to replace all usage of it. Though it would require new release of kogito anyway so how urgent is to unblock the quarkus issue 6119?

**Comment by dmlloyd** [comment]
I was hoping to get that upgrade in before the imminent 1.1.0 Quarkus release; however, I'm not sure if @gsmet is going to be keen on requiring a Kogito upgrade to proceed (the 1.1.0 release is already burdened with a few problems).  So: I'm not certain. :slightly_smiling_face:

**Comment by gsmet** [comment]
I could be convinced to upgrade Kogito in Final if the changes are minimal and it doesn't end up being a rabbit hole.

**Comment by mswiderski** [comment]
so what's the time frame for Final? depending on the nexus and sync with maven central it could take few days to get the kogito released...

**Comment by gsmet** [comment]
It was supposed to be on Monday but given CR1 had a delay, I think Tuesday will be better.

**Comment by mswiderski** [comment]
created kogito jira for it and PR is already prepared https://issues.redhat.com/browse/KOGITO-770, let's see when we can get this done and try to spin off 0.6.1 release

**Comment by mswiderski** [comment]
this has been addressed in 0.6.1 of kogito

**Comment by dmlloyd** [comment]
Thanks!
